app/controllers/application_controller.rb

```diff
@@ -348,13 +348,11 @@ def check_order_id
   end
 
   def reset_session_ids
-    if !session[:order_id].nil?
-      session[:order_id] = nil
-      session[:current_delivery_id] = nil
-      session[:current_organization_id] = nil
-      session[:current_supplier_id] = nil
-      session[:cart_id] = nil
-    end
+    session[:order_id] = nil
+    session[:current_delivery_id] = nil
+    session[:current_organization_id] = nil
+    session[:current_supplier_id] = nil
+    session[:cart_id] = nil
```

Comment:
@thermistor @ivar this change makes sense from the method name, but worries me a bit. I'm wondering if there are any unintended side effects of now resetting `:current_delivery_id`, `:current_organization_id`, `:current_supplier_id`, and `:cart_id` regardless of `:order_id` being not nil. Was this just some code cleanup? If it was pure cleanup, I'd say we revert and put into a JIRA task and get some spec coverage of this.

Reply:
This method `reset_session_ids` is only called from `check_order_id` line 343

Reply:
Oh okay, looks good then.